src/drivers/gps/gps.cpp

```diff
@@ -1361,9 +1361,14 @@ GPS *GPS::instantiate(int argc, char *argv[], Instance instance)
 		return nullptr;
 	}
 
-	GPS *gps;
+	GPS *gps = nullptr;
 	if (instance == Instance::Main) {
-		gps = new GPS(device_name, mode, interface, enable_sat_info, instance, baudrate_main);
+		if (access(device_name, R_OK|W_OK) == 0) {
+			gps = new GPS(device_name, mode, interface, enable_sat_info, instance, baudrate_main);
+
+		} else {
+			PX4_ERR("invalid device (-d) %s", device_name);
```

Comment:
Needs a nullptr check, and I'd add it before `access` as well.

Reply:
Done.